sourcerer-dbstore/src/main/java/org/elder/sourcerer2/dbstore/DbstoreEventRepository.kt

```diff
@@ -0,0 +1,254 @@
+                is DbstoreEventRow.Event -> {
+                    events.add(fromDbstoreEventRecord(row.eventData))
+                    lastVersion = versionExtractor(row.eventData)
+                }
+                is DbstoreEventRow.EndOfStream -> {
+                    foundEndOfStream = true
+                    break@scanrows
+                }
+            }
+        }
+
+        return when {
+            lastVersion != null -> {
+                // Happy path, we have read some rows - doesn't matter if we had a version specified or not, the new
+                // version is the one of the last event read
+                logger.debug(
+                        "Read stream {} up to version {}",
+                        streamIdentifier, lastVersion
+                )
+                resultCreator(ImmutableList.copyOf(events), foundEndOfStream, lastVersion)
+            }
+            foundEndOfStream && version != null -> {
+                // We got no events but had a "from" specified, we assume the caller knows what they're talking
+                // about (the version does exist in this stream) and there are no newer events, so the version
+                //  specified is still relevant.
+                logger.debug(
+                        "Read {} - with no new events after {}",
+                        streamIdentifier, lastVersion
+                )
+
+                resultCreator(ImmutableList.copyOf(events), foundEndOfStream, version)
+            }
+            else -> {
+                // We have read no events, and found no end of stream marker, so the stream can't exist.
+                // Check for some potential weird cases that should not happen, e.g. we have an end of stream marker
+                // but no events.
+                if (foundEndOfStream) {
+                    logger.error(
+                            "Read {} from beginning and found sentinel but no events!",
+                            streamIdentifier
+                    )
+                }
+
+                logger.debug(
+                        "Read {} but found nothing",
+                        streamIdentifier, lastVersion
+                )
+                null
+            }
+        }
+    }
+
+    override fun readFirst(streamId: StreamId): EventRecord<T> {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun readLast(streamId: StreamId): EventRecord<T> {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun getCurrentVersion(): RepositoryVersion {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun getCurrentVersion(streamId: StreamId): StreamVersion {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Line detected that is longer than the defined maximum line length in the code style. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3AMaxLineLength)